mlflow/transformers.py

```diff
@@ -0,0 +1,851 @@
+import contextlib
+import importlib
+import logging
+import pathlib
+
+from typing import Union, List, Optional, Dict, Any
+import yaml
+import warnings
+
+import mlflow
+from mlflow import pyfunc
+from mlflow.exceptions import MlflowException
+from mlflow.models import ModelInputExample, Model, infer_pip_requirements
+from mlflow.models.model import MLMODEL_FILE_NAME
+from mlflow.models.signature import ModelSignature
+from mlflow.models.utils import _save_example
+from mlflow.protos.databricks_pb2 import INVALID_PARAMETER_VALUE, BAD_REQUEST
+from mlflow.tracking._model_registry import DEFAULT_AWAIT_MAX_SLEEP_SECONDS
+from mlflow.utils.annotations import experimental
+from mlflow.utils.docstring_utils import format_docstring, LOG_MODEL_PARAM_DOCS
+from mlflow.utils.environment import (
+    _mlflow_conda_env,
+    _validate_env_arguments,
+    _CONDA_ENV_FILE_NAME,
+    _PYTHON_ENV_FILE_NAME,
+    _process_conda_env,
+    _process_pip_requirements,
+    _CONSTRAINTS_FILE_NAME,
+    _REQUIREMENTS_FILE_NAME,
+    _PythonEnv,
+)
+from mlflow.utils.file_utils import write_to
+from mlflow.utils.model_utils import (
+    _validate_and_copy_code_paths,
+    _validate_and_prepare_target_save_path,
+    _download_artifact_from_uri,
+    _get_flavor_configuration_from_uri,
+    _add_code_from_conf_to_system_path,
+)
+from mlflow.utils.requirements_utils import _get_pinned_requirement
+
+FLAVOR_NAME = "transformers"
+_PIPELINE_BINARY_KEY = "pipeline"
+_PIPELINE_BINARY_FILE_NAME = "pipeline.tr"
+_COMPONENTS_BINARY_KEY = "components"
+_MODEL_KEY = "model"
+_TOKENIZER_KEY = "tokenizer"
+_FEATURE_EXTRACTOR_KEY = "feature_extractor"
+_IMAGE_PROCESSOR_KEY = "image_processor"
+_PROCESSOR_KEY = "processor"
+_TOKENIZER_TYPE_KEY = "tokenizer_type"
+_FEATURE_EXTRACTOR_TYPE_KEY = "feature_extractor_type"
+_IMAGE_PROCESSOR_TYPE_KEY = "image_processor_type"
+_PROCESSOR_TYPE_KEY = "processor_type"
+_CARD_TEXT_FILE_NAME = "model_card_text.txt"
```

Comment:
Can we use markdown here? Model cards seem to use markdown format, since they're defined by README.md:

```
What are Model Cards?
Model cards are files that accompany the models and provide handy information. Under the hood, model cards are simple Markdown files with additional metadata. Model cards are essential for discoverability, reproducibility, and sharing! You can find a model card as the README.md file in any model repo.
```

```suggestion
_CARD_TEXT_FILE_NAME = "model_card_text.md"
```

Reply:
Can we simply call this file `model_card.md`?

Reply:
changed to model_card.md